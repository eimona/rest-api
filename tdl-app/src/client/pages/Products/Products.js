import React from 'react';

import { Grid } from '@material-ui/core';

import ProductTile from '../../components/ProductTile';

const exampleData = [
    {
        productCode: '01',
        productName: 'Super Amazing Awsome Cake',
        sellerName: 'Gordon Ramsay',
        productPrice: '$123.45',
        images: 'https://images.immediate.co.uk/production/volatile/sites/2/2019/04/Choc-Fudge-Cake-b2d1909.jpg?quality=45&resize=620,413',
        productDescription: 'This cake is super duper amazing!'
    },
    {
        productCode: '02',
        productName: 'Not So Amazing Cake',
        sellerName: 'El Goron Ramsi',
        productPrice: '$67.89',
        images: 'https://7yearolds.com/wp-content/uploads/2018/11/cake.jpg',
        productDescription: 'This cake is not so amazing...'
    },
];

const Products = () => {
    const params = new URLSearchParams(window.location.search);
    const category = params.get('category');
    return (
        <div>
            <p>
                {category}
            </p>
            <Grid
            container
            spacing={5}
            direction="column"
            alignItems="center"
            justify="center"
            style={{ minHeight: '90vh' }}
            >
                {exampleData.map(detail => (
                    <Grid item>
                        <ProductTile productDetails={detail}/>
                    </Grid>   
                ))}
            </Grid>
        </div>
    );
};

export default Products;
