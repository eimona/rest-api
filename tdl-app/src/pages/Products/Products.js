import React from 'react';

const Products = () => {
    return (
        <div>
            <p>a</p>
        </div>
    );
};

export default Products;
