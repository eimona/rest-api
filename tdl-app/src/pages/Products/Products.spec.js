import React from 'react';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom'

import Products from './Products';

describe('Products', () => {
    it('renders correctly', () => {
        const tree = renderer
          .create(
              <MemoryRouter>
                  <Products />
              </MemoryRouter>
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
});
