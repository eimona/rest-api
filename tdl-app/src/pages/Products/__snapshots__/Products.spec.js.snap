// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products renders correctly 1`] = `
<div>
  <p />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    style={
      Object {
        "minHeight": "90vh",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root"
        href="/detailedProduct?productCode=01"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-104 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Super Amazing Awsome Cake
            </h2>
            <h2
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Gordon Ramsay
            </h2>
            <h2
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              $123.45
            </h2>
            <div
              className="MuiCardMedia-root makeStyles-media-105"
              style={
                Object {
                  "backgroundImage": "url(\\"https://images.immediate.co.uk/production/volatile/sites/2/2019/04/Choc-Fudge-Cake-b2d1909.jpg?quality=45&resize=620,413\\")",
                }
              }
              title="Product"
            />
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              This cake is super duper amazing!
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root"
        href="/detailedProduct?productCode=02"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-104 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Not So Amazing Cake
            </h2>
            <h2
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              El Goron Ramsi
            </h2>
            <h2
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              $67.89
            </h2>
            <div
              className="MuiCardMedia-root makeStyles-media-105"
              style={
                Object {
                  "backgroundImage": "url(\\"https://7yearolds.com/wp-content/uploads/2018/11/cake.jpg\\")",
                }
              }
              title="Product"
            />
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              This cake is not so amazing...
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
